drivers/flash/spi_nor.c

```diff
@@ -427,11 +442,13 @@ static int enter_dpd(const struct device *const dev)
 static int exit_dpd(const struct device *const dev)
 {
 	int ret = 0;
+	const struct spi_nor_config *cfg = dev->config;
 
-	if (IS_ENABLED(DT_INST_PROP(0, has_dpd))) {
+	ARG_UNUSED(cfg);
+	if (IS_ENABLED(DT_INST_PROP(cfg->index, has_dpd))) {
 		delay_until_exit_dpd_ok(dev);
 
-#if DT_INST_NODE_HAS_PROP(0, dpd_wakeup_sequence)
+#if ANY_INST_HAS_DPD_WAKEUP_SEQUENCE
```

Comment:
I've added the dev review tag to this PR - the dev review meeting is today, if you're able to join and discuss this then great, if not then I will bring it up and ask for feedback on the best way to implement this

Reply:
I could use the following construct `if (IS_ENABLED(DT_INST_PROP(instance_indx, opt-in-prop)))`
Is that a good solution?

Reply:
On https://lists.zephyrproject.org/ as part of the devel group, there will be an email sent with the joining link (teams meeting). It is at 8:00am (UTC-7)

Reply:
Maybe this works to solve the issue and avoid knowing what is a correct "default value" ?

- Setup the exist flag (TODO)
   `COND_CODE_1(INST_HAS_PROP, (.prop_exist = true), (.prop_exist = false))`

- Setup the value (this already done in the current commit)
  `COND_CODE_1(INST_HAS_PROP, (.prop_value = <DTS>), (.prop_value = 0))`

- Code is surrounded by `#if ANY_INST...` to get compiled (already done)

- Code is conditional on `if prop_exist` to be executed (TODO)

Reply:
> 

Yeah, but we need to figure out the flags in some smart way, otherwise you will end up with bunch of bool variables, that I am not sure will get packed.

Reply:
Can we agree on a list of properties that need this? because all the DTS props are opt-in (as far as I can tell at least)

These are boolean flags:
requires-ulbpr
has-lock

These can be regrouped in one flag because they are related:
has-dpd
t-enter-dpd
t-exit-dpd

These will need an exist flag:
dpd-wakeup-sequence
mxicy,mx25r-power-mode
enter-4byte-addr

These are not used in code:
quad-enable-requirements
wp-gpios
hold-gpios
reset-gpios

I don't know about these:
jedec-id
size
sfdp-bfp

Reply:
@Laczen , @de-nordic , @nordicjm , can you please help me with the failing CI test?
The CI reports this test as failing:  
`tests/lib/mpsc_pbuf/libraries.mpsc_pbuf_concurrent on qemu_x86_64 failed (unexpected eof)`

I tried the test on a fresh workspace and the test passes
Should I just rebase the PR? or is there something I'm not seeing? 
Thank you for your help

Environment informations: 
Git head: 0e6c306dcec991311b11b4090430ade43d04777e

Git status output:
```
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   ../drivers/flash/spi_nor.c

no changes added to commit (use "git add" and/or "git commit -a")
```

Build output:
```
west build -p -b qemu_x86_64 -T tests/lib/mpsc_pbuf/libraries.mpsc_pbuf_concurrent
-- west build: making build dir /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/tests/lib/mpsc_pbuf
-- CMake version: 3.26.2
-- Found Python3: /usr/local/opt/python@3.11/bin/python3.11 (found suitable exact version "3.11.2") found components: Interpreter 
-- Cache files will be written to: /Users/mehdizemzem/Library/Caches/zephyr
-- Zephyr version: 3.3.99 (/Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr)
-- Found west (found suitable version "1.0.0", minimum required is "0.7.1")
-- Board: qemu_x86_64
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.0 (/Users/mehdizemzem/zephyr_fork/zephyr-sdk-0.16.0)
-- Found toolchain: zephyr 0.16.0 (/Users/mehdizemzem/zephyr_fork/zephyr-sdk-0.16.0)
-- Found Dtc: /usr/local/bin/dtc (found suitable version "1.7.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/boards/x86/qemu_x86/qemu_x86_64.dts
-- Generated zephyr.dts: /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build/zephyr/dts.cmake
Parsing /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/Kconfig
Loaded configuration '/Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/boards/x86/qemu_x86/qemu_x86_64_defconfig'
Merged configuration '/Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/tests/lib/mpsc_pbuf/prj.conf'
Merged configuration '/Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build/zephyr/misc/generated/extra_kconfig_options.conf'
Configuration saved to '/Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build/zephyr/.config'
Kconfig header saved to '/Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build/zephyr/include/generated/autoconf.h'
-- Found GnuLd: /Users/mehdizemzem/zephyr_fork/zephyr-sdk-0.16.0/x86_64-zephyr-elf/bin/../lib/gcc/x86_64-zephyr-elf/12.2.0/../../../../x86_64-zephyr-elf/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /Users/mehdizemzem/zephyr_fork/zephyr-sdk-0.16.0/x86_64-zephyr-elf/bin/x86_64-zephyr-elf-gcc
CMake Warning at /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/subsys/random/CMakeLists.txt:11 (message):
  

      Warning: CONFIG_TIMER_RANDOM_GENERATOR is not a truly random generator.
      This capability is not secure and it is provided for testing purposes only.
      Use it carefully.


CMake Warning at /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/CMakeLists.txt:829 (message):
  No SOURCES given to Zephyr library: lib__libc__common

  Excluding target from build.


-- Configuring done (3.8s)
-- Generating done (0.2s)
-- Build files have been written to: /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/build
-- west build: building application
[1/135] Preparing syscall dependency handling

[2/135] Generating include/generated/version.h
-- Zephyr version: 3.3.99 (/Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr), build: zephyr-v3.3.0-1928-g0e6c306dcec9
[125/135] Linking C executable zephyr/zephyr_pre0.elf

[129/135] Linking C executable zephyr/zephyr_pre1.elf

[135/135] Linking C executable zephyr/zephyr.elf
Memory region         Used Size  Region Size  %age Used
             RAM:        264 KB         3 MB      8.59%
          LOCORE:         32 KB        60 KB     53.33%

```

Twister output:
```
west twister -p qemu_x86_64 -s tests/lib/mpsc_pbuf/libraries.mpsc_pbuf_concurrent
Renaming output directory to /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/twister-out.4
INFO    - Using Ninja..
INFO    - Zephyr version: zephyr-v3.3.0-1928-g0e6c306dcec9
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/twister-out/testplan.json
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete:    1/   1  100%  skipped:    0, failed:    0, error:    0
INFO    - 1 test scenarios (1 test instances) selected, 0 configurations skipped (0 by static filter, 0 at runtime).
INFO    - 1 of 1 test configurations passed (100.00%), 0 failed, 0 errored, 0 skipped with 0 warnings in 45.25 seconds
INFO    - In total 21 test cases were executed, 3 skipped on 1 out of total 554 platforms (0.18%)
INFO    - 1 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/twister-out/twister.json
INFO    - Writing xunit report /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /Users/mehdizemzem/zephyr_fork/zephyr_dev/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

Reply:
It is a flaky CI test, have reran it.

Reply:
@ashiroji  Sorry, I did not found time to take deeper look into this. I will try to this week.

Reply:
@de-nordic , @nordicjm still waiting for review please.

Reply:
You need to re-request reviews in the top right of a PR to flag it so others can see they need to re-review. That being said, you have a merge conflict which needs resolving.